File: frontend/src/types/stop.ts
export interface StopResponse {
  stop_id: string;
  name: string;
  lat: number;
  lon: number;
  sequence: number;
  arrival_time: string | null;
  departure_time: string | null;
}

export interface StopTimeResponse {
  route_id: string;
  route_short_name: string;
  trip_id: string;
  headsign: string | null;
  arrival_time: string;
}

File: backend/app/schemas/stop.py
from pydantic import BaseModel


class StopResponse(BaseModel):
    stop_id: str
    name: str
    lat: float
    lon: float
    sequence: int
    arrival_time: str | None = None
    departure_time: str | None = None


class StopTimeResponse(BaseModel):
    route_id: str
    route_short_name: str
    trip_id: str
    headsign: str | None = None
    arrival_time: str
